Cue/app/tabs/library/DeckSharingOptions.js

```diff
@@ -0,0 +1,391 @@
+      : deck.share_code
+        ? 'shared'
+        : 'private'
+  }
+
+  _arraysEqual = (a: ?Array<*>, b: ?Array<*>): boolean => {
+    if (!a || !b) {
+      return false
+    }
+    
+    if (a.length !== b.length) {
+      return false
+    }
+
+    for (let i = 0; i < a.length; i++) {
+      if (a[i] !== b[i]) {
+        return false
+      }
+    }
+
+    return true
+  }
+
+  /* ==================== Edit Flow ==================== */
+  _onSubmit = () => {
+    if (this.state.selectedOption === 'public') {
+      this._handlePublicSubmit()
+    } else if (this.state.selectedOption === 'shared') {
+      this._handleSharedSubmit()
+    } else if (this.state.selectedOption === 'private') {
+      this._handlePrivateSubmit()
+    }
+  }
+
+  _handlePublicSubmit = () => {
+    let oldTags = this.props.deck.tags
+    let newTags = this.state.tags
+
+    if (!this._arraysEqual(oldTags, newTags)) {
+      this.props.editTags(this.props.deck.uuid, newTags)
+    }
+
+    if (!this.props.deck.public) {
+      this.props.setPublic(this.props.deck.uuid, true)
+    }
+
+    this.props.navigator.pop()
+  }
+
+  _handleSharedSubmit = () => {
+    // Generate a share code if needed
+    if (!this.props.deck.share_code) {
+      LibraryApi.generateShareCode(this.props.deck.uuid)
+        .then(response => {
+          this.setState({
+            dirty: false,
+            shareCode: response.share_code
+          })
+          this.props.setPublic(this.props.deck.uuid, false, false)
+          this.props.recordShareCode(this.props.deck.uuid, response.share_code)
+        })
+        .catch(e => {
+          console.warn('Failed to generate share code', e)
+          Alert.alert(
+            (Platform.OS === 'android'
+              ? 'Could not generate share code'
+              : 'Could Not Generate Share Code'),
+            'Check your Internet connection and try again.',
+            [{text: 'OK', style: 'cancel'}]
+          )
+        })
+    } else {
+      // Set public to false to restrict the deck to Shared status
+      if (this.props.deck.public) {
+        this.props.setPublic(this.props.deck.uuid, false, false)
+      }
+
+      this.props.navigator.pop()
+    }
+  }
+
+  _handlePrivateSubmit = () => {
+    let message
+    switch (this._sharingStatusForDeck(this.props.deck)) {
+      case 'public':
+        message = 'Your deck will no longer appear in Discover or Search.'
+          + '\n\nPeople who currently have your deck in their Library will still be able to make copies of your deck.'
+        break
+      case 'shared':
+        message = 'The share code for your deck will become invalid immediately.'
+          + '\n\nPeople who currently have your deck in their Library will still be able to make copies of your deck.'
+        break
+    }
+
+    if (message) {
+      Alert.alert(
+        (Platform.OS === 'android'
+          ? 'Others may lose access to this deck'
+          : 'Others May Lose Access to this Deck'),
```

Comment:
Yeah, alert titles are generally written in title case except in the rare case that they're formulated as complete sentences, in which case they're written using sentence case.